src/execution_engines/Engine.ts

```diff
@@ -0,0 +1,27 @@
+export interface LoadContractIntoEngine {
```

Comment:
I don't think type should be used except for unions etc. TS recommend the use of `interface` over `type`: https://github.com/Microsoft/TypeScript/blob/master/doc/spec.md#3.10

"Interface types have many similarities to type aliases for object type literals, but since interface types offer more capabilities they are generally preferred to type aliases"

Reply:
Agree with renaming these interfaces though